### ISSUE TITLE
Failure to program nspanel_blank.tft on Nextion loaded with nspanel-lovelace-ui

### ISSUE THREAD

**Issue by jvanstraten** [Blackymas/NSPanel_HA_Blueprint#1374]
### tl;dr

I'm pretty sure I've already followed all the [troubleshooting steps](https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-TFT-Upload) as far as they apply, and I _am_ trying to [load nspanel_blank.tft first](https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-TFT-Upload-When-nspanel-lovelace-ui-has-been-installed), but the display doesn't seem to be going into programming mode at all. I went all the way to hooking up a logic analyzer to see what the ESP and display are doing, but didn't really get anywhere.

### Context

I'm almost completely new to HomeAssistent, and am completely new to ESPHome. Thus far I've just been making and programming my own microcontroller + relay boards. My only prior experience is from a few months ago, when I first tried to get my NSPanel to do something. I went for the Tasmota + lovelace UI TFT file, but ultimately tossed it in a corner for half a year because it wasn't doing what I wanted. I'm not sure this project is doing that either (I'm looking for something lightweight that I can use as baseline for my own C programming, in particular because I was hoping to use some spare I/Os on the NSPanel to connect to my homebrew serial network), but regardless, I figured I'd give it a try. I'm afraid I don't know exactly which commit at [joBr99/nspanel-lovelace-ui](https://github.com/joBr99/nspanel-lovelace-ui) I stuck on there, but I was working on this mid August so it'd be some commit around that time, and I *do* know I never got around to changing HMI files.

### What I did

 - Followed the steps up to 6 from [here](https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-First-Steps---Installation-and-Setup), then followed the video and, after some confusion, managed to flash ESPHome. YAML, minus comments:

```
substitutions:
  device_name: "nskitchen"
  wifi_ssid: !secret wifi_ssid
  wifi_password: !secret wifi_password
  nextion_update_url: "http://homeassistant.local:8123/local/nspanel_eu.tft"

packages:
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    files:
      - nspanel_esphome.yaml
    refresh: 300s
esp32:
  framework:
    type: esp-idf
```

 - Compilation was a few hours ago, so probably f357a58b9 (v4.1.2). 
 - Step 7 of the above seemed to be automatic, or otherwise I'm not understanding what it means; in any case, I didn't do anything here. I don't think it should have any bearing on flashing the Nextion, but just in case.
 - Instead of step 8, I followed the [nspanel_blank.tft](https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-TFT-Upload-When-nspanel-lovelace-ui-has-been-installed) instructions as instructed at the top of the page. In case it matters for some reason, I was initially lazy by just renaming nspanel_blank.tft to nspanel_eu.tft and then starting the default display update, but in the end actually did it properly as I started grasping at straws. All TFT files are also from f357a58b9 (v4.1.2).

### What went wrong

The Nextion never seems to enter its programming mode; it keeps showing the lovelace UI "no response from HomeAssistant yet" spinner. That spinner *does* start lagging a whole lot, its refresh rate reducing to the rate at which the ESP is spamming `[xx:xx:xx][W][nextion:077]: Nextion is not connected!`. After letting it run for a few minutes, I decided it's probably not working, so I power cycled. The spinner comes right back up again, so if it _did_ program anything, it couldn't have been much.

### What I already tried

From the [troubleshooting](https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-TFT-Upload) page:

1) **Using a local source.** I already am. Also checked to make sure behavior is different when the file is *not* being served by HA.
2) **HTTP instead of HTTPS.** Seems N/A.
3) **Network issues.** I doubt this has anything to do with it, everything else on the network seems stable.
4) **DNS.** Seems N/A.
5) **Check the file size by downloading to your computer.** Way ahead of you, I already did an md5sum check just to be sure.
6) **Power cycle your panel.** I tried several times.
7) **Flash your panel again.** I didn't have any customizations to begin with; everything, including HA itself in fact, is a fresh install. Since the device showed up in ESPHome just fine, I'm not sure what flashing the same thing again would accomplish in my case.
8) **Alternative http service.** I didn't try this, because as part of step 1 I'm already pretty sure the download is not the problem.

I tried to look for similar issues, but there are so many of them that I quickly got lost. This part of the log ([full log here](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/13539777/logs_nskitchen_logs.txt)) seems different from the issues I've seen:

```
[18:07:36][D][nextion_upload:130]: Nextion TFT upload requested
[18:07:36][D][nextion_upload:131]: url: http://homeassistant.local:8123/local/nspanel_blank.tft
[18:07:36][D][nextion_upload:178]: TFT file size: 275288
[18:07:36][D][nextion_upload:190]: Updating Nextion
[18:07:38][D][nextion_upload:218]: Upgrade response is []                <========= ???
[18:07:38][E][nextion_upload:223]: Preparation for tft update failed 0 ""
[18:07:38][D][nextion_upload:254]: Restarting Nextion
```

To further debug, I tapped into the serial port between the ESP32 and the Nextion with my logic analyzer by soldering to the respective test points. [Data attached](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/13539782/nskitchen-data.zip); the .sal file can be viewed with [Logic](https://www.saleae.com/downloads/) regardless of whether you have a Saleae device or not. I'm not sure what's going on with this link because I've yet to dive into the Nextion docs. I guess that's next on my todo list (that, or buying a new NSPanel...), but I figured I'd ask here, too.

Any ideas on where to go from here?

**Comment by edwardtfn** [comment]
> Compilation was a few hours ago, so probably v4.1.2

We have release v4.1.3 a couple of hours ago, so if you want update it, it might be a good idea. You dint have to go the whole process anymore, as you may be able to see the panel in your ESPHome dashboard, so just click the 3-dot menu, select "Clean build files", then click "Install" and select the option to install it wirelessly, so you dint have to disassemble your panel again.

I will come with some more comments to your message in a minute,

**Comment by edwardtfn** [comment]
> All TFT files are also from https://github.com/Blackymas/NSPanel_HA_Blueprint/commit/f357a58b9ee9b88f2eead61f3e7a86264e020cc1 (v4.1.2).

That's fine. No need to update those TFT files.

**Comment by edwardtfn** [comment]
> Alternative http service. I didn't try this, because as part of step 1 I'm already pretty sure the download is not the problem.

> `[18:07:36][D][nextion_upload:131]: url: http://homeassistant.local:8123/local/nspanel_blank.tft`

I don't think this is the problem. I can see in the logs that the panel could get a file from your server, so it looks OK.

**Comment by edwardtfn** [comment]
> keeps showing the lovelace UI "no response from HomeAssistant yet" spinner. That spinner does start lagging a whole lot, its refresh rate reducing to the rate at which the ESP is spamming

Please click times in the center of that ring until a new page is shown. On that page, please click the buttons to set recmod and set baud rate 115200. I don't remember exactly the labels, but both should be in the bottom right corner of the screen (at least it is in the EU version).

**Comment by jvanstraten** [comment]
I checked with my LA that both sides are chatting at 115200, at least at a glance. But that lovelace-ui debug page looks promising; I'll retry.

**Comment by jvanstraten** [comment]
Ok, gotcha, I wasn't sure from your message if it would affect the upload process. Redoing.

Phenomenal support response time, by the way. Much appreciated!
